profiles/views.py

```diff
@@ -39,19 +38,79 @@ def PullRequest(request):
             leaderboard = Leaderboard.objects.get(username=user)
         except:
             return Response(
-            {"detail":"Cannot retrieve the user."}, 
-            status=status.HTTP_404_NOT_FOUND
-        )
+                {"detail":"Cannot retrieve the user."},
+                status=status.HTTP_404_NOT_FOUND
+            )
         else:
             if action == 'opened':
-                leaderboard.pr_opened +=1
+                leaderboard.pr_opened += 1
                 leaderboard.save()
             elif action == 'closed' and merged:
-                leaderboard.pr_merged +=1
+                leaderboard.pr_merged += 1
                 leaderboard.save()
-            else: 
+            else:
                 pass
             return Response(
                 {'detail':'Successfully updated leaderboard'},
                 status=status.HTTP_200_OK
             )
+
+@swagger_auto_schema(
+        operation_id='update pull requests status',
+        method='post',
+        responses={
+            '200': set_example({'detail':'Successfully updated leaderboard'}),
+            '400': set_example({"detail":"Sorry, there is some issue with the webhooks."}),
+            '404': set_example({"detail":"Cannot retrieve the user."})
+        },
+)
+@api_view(['post'])
+def issue(request):
+    """
+    This API is to keep a track of the issues closed and the \
+    contribution the user assigned to close it. This is \
+    automatically handled by webhooks in the git repository \
+    or repositories being tracked.
+    """
+
+    try:
+        action = request.data["action"]
+        labels = request.data["issue"]["labels"]
+        username = request.data["issue"]["assignee"]["login"]
```

Comment:
```suggestion
        assignee = request.data["issue"]["assignee"]
```
This is because when the issue isn't assigned to anyone. there isnt any `'login'` key and hence raises a `HTTP_400_BAD_REQUEST` .

I think if it says that the user could not be found. :v:

Reply:
Nice catch! Gotcha 🔥